### PR TITLE
Fix the problem with double clientids (now for real)

### PR DESCRIPTION
**Pull request by Gabgab2003**
This uses redis sadd to make a set called client_ids and writes users into it when they are created. Someone should probably make something that deletes users when the connection (chunked http stream) ends.
Open for better Ideas

### CODE REVIEW

File: server.rb [hunk 1]
```diff
@@ -10,9 +10,12 @@
 NEW_MESSAGE_CHANNEL = 'new_message_channel'.freeze
 UPDATED_CLIENT_CHANNEL = 'updated_client_channel'.freeze
 MESSAGE_LIST_KEY = 'message_list'.freeze
+CLIENT_IDS_KEY = 'client_ids'.freeze
 
 # Clear out any old messages when we boot up
 Redis.new(url: ENV['REDIS_URL']).del(MESSAGE_LIST_KEY)
+Redis.new(url: ENV['REDIS_URL']).del(CLIENT_IDS_KEY)
```
Comment (kkuchta) [inline server.rb]: Might as well reuse the connection here:
redis = Redis.new(url: ENV['REDIS_URL'])
redis.del(MESSAGE_LIST_KEY)
redis.del(CLIENT_IDS_KEY)

File: server.rb [hunk 2]
```diff
@@ -103,9 +106,13 @@ def redis
...
 
   def each(&each_block)
-    # Generate a random name to differentiate clients  Duplicates will break
-    # everything, but ¯\_(ツ)_/¯
+    # Generate a random name to differentiate clients
+    # If a name already exists, the name is rerolled
+    begin
     client_id = Faker::Name.first_name + rand(1000).to_s
+    end while Redis.new(url: ENV['REDIS_URL']).sismember(CLIENT_IDS_KEY, client_id)
+    Redis.new(url: ENV['REDIS_URL']).sadd(CLIENT_IDS_KEY, client_id)
```
Comment (kkuchta) [inline server.rb]: On both 113 and 114, you can just use the IndexStreamer object's existing `redis` method (like we do down in the `messages_html` method):
```
    begin
      client_id = Faker::Name.first_name + rand(1000).to_s
    end while redis.sismember(CLIENT_IDS_KEY, client_id)
    redis.sadd(CLIENT_IDS_KEY, client_id)
```

Comment (Gabgab2003) [reply server.rb]: @kkuchta I'll free some time in my schedule today to work your changes in :+1:

**Comment by Gabgab2003** [comment]
Ive worked in your changes

**Comment by Gabgab2003** [comment]
@kkuchta This should be merge ready

**Comment by kkuchta** [comment]
Looks good to me!  Nice fix.  :)

**Pull request merged**